docs/api-inference/tasks/image-text-to-text.md

```diff
@@ -0,0 +1,129 @@
+<!---
+This markdown file has been generated from a script. Please do not edit it directly.
+For more details, check out:
+- the `generate.ts` script: https://github.com/huggingface/hub-docs/blob/main/scripts/api-inference/scripts/generate.ts
+- the task template defining the sections in the page: https://github.com/huggingface/hub-docs/tree/main/scripts/api-inference/templates/task/image-text-to-text.handlebars
+- the input jsonschema specifications used to generate the input markdown table: https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/image-text-to-text/spec/input.json
+- the output jsonschema specifications used to generate the output markdown table: https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/image-text-to-text/spec/output.json
+- the snippets used to generate the example:
+  - curl: https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/snippets/curl.ts
+  - python: https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/snippets/python.ts
+  - javascript: https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/snippets/js.ts
+- the "tasks" content for recommended models: https://huggingface.co/api/tasks
+--->
+
+## Image-Text to Text
+
+Image-text-to-text models take in an image and text prompt and output text. These models are also called vision-language models, or VLMs. The difference from image-to-text models is that these models take an additional text input, not restricting the model to certain use cases like image captioning, and may also be trained to accept a conversation as input.
+
+<Tip>
+
+For more details about the `image-text-to-text` task, check out its [dedicated page](https://huggingface.co/tasks/image-text-to-text)! You will find examples and related materials.
+
+</Tip>
+
+### Recommended models
+
+- [HuggingFaceM4/idefics2-8b-chatty](https://huggingface.co/HuggingFaceM4/idefics2-8b-chatty): Cutting-edge conversational vision language model that can take multiple image inputs.
+- [microsoft/Phi-3.5-vision-instruct](https://huggingface.co/microsoft/Phi-3.5-vision-instruct): Strong image-text-to-text model.
+
+This is only a subset of the supported models. Find the model that suits you best [here](https://huggingface.co/models?inference=warm&pipeline_tag=image-text-to-text&sort=trending).
+
+### Using the API
+
+
+<inferencesnippet>
+
+<curl>
+```bash
+curl https://api-inference.huggingface.co/models/HuggingFaceM4/idefics2-8b-chatty \
+	-X POST \
+	-d '{"inputs": No input example has been defined for this model task.}' \
+	-H 'Content-Type: application/json' \
+	-H "Authorization: Bearer hf_***"
+```
+</curl>
+
+<python>
+```py
+import requests
+
+API_URL = "https://api-inference.huggingface.co/models/HuggingFaceM4/idefics2-8b-chatty"
+headers = {"Authorization": "Bearer hf_***"}
+
+from huggingface_hub import InferenceClient
```

Comment:
This is not expected (i.e. having `import requests ...` before `from huggingface_hub import InferenceClient`). I realized that https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct?inference_api=true has a problem. Model doesn't have a chat template and therefore is not tagged as `"conversational"` which creates this weird side effect.

So I see 3 independent things to correct here:
1. it would be nice to recommend `meta-llama/Llama-3.2-11B-Vision-Instruct` first on the image-text-to-image [task page](https://huggingface.co/tasks/image-text-to-text) (to update [here](https://github.com/huggingface/huggingface.js/blob/a22a62a0cfc9fa644a9d40e4bbfcd0489cc11038/packages/tasks/src/tasks/image-text-to-text/data.ts#L44))
2. we should fix in moon-landing the `"conversational"` detection. At the moment, it's based only on the presence of a chat template. However for idefics chatty 8b it seems it's using ["use_default_system_prompt": true](https://huggingface.co/HuggingFaceM4/idefics2-8b-chatty/blob/main/tokenizer_config.json#L70) instead. @Rocketknight1 is it safe to assume that a model with no chat template but this parameter set to True is in fact a conversational model? And if not, which parameter could we check?
3. For non-conversational `image-text-to-text` (does that even exist?), we should fix the [snippet generator](https://github.com/huggingface/huggingface.js/blob/a22a62a0cfc9fa644a9d40e4bbfcd0489cc11038/packages/tasks/src/snippets/python.ts#L184) so that only the `requests`-based snippet is displayed instead of this weird combination.

cc @osanseviero as well for viz'

Reply:
just to add, `HuggingFaceM4/idefics2-8b-chatty` has the`chat_template` defined in the [processor_config.json](https://huggingface.co/HuggingFaceM4/idefics2-8b-chatty/blob/main/processor_config.json). The `tokenizer.chat_template` attribute is supposed to be saved in `tokenizer_config.json` file, maybe it's something specific to this model.

Reply:
for the 3rd point, pinging @mishig25 since it's related to [huggingface.js/pull/938](https://github.com/huggingface/huggingface.js/pull/938). do you think it's okay to map [image-text-to-text](https://github.com/huggingface/huggingface.js/blame/22e6d4dc924fd9f876fae2dbbea8e4330cd282eb/packages/tasks/src/snippets/python.ts#L158) to `snippetBasic` instead and define the task input [here](https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/snippets/inputs.ts) ?